carbonmark-api/src/routes/projects/get.ts

```diff
@@ -52,280 +54,93 @@ const schema = {
-      project.name = cmsData ? cmsData.name : project.name;
-      project.methodologies = cmsData ? cmsData.methodologies : [];
 
-      project.short_description = cmsData?.projectContent
-        ? cmsData.projectContent.shortDescription
-        : undefined;
-      project.long_description = cmsData?.projectContent
-        ? cmsData.projectContent.longDescription
-        : undefined;
+      // Find the lowest price
+      // @todo change to number[]
+      const listingPrices = compact(project.listings)
+        ?.filter(isListingActive)
+        .map(extract("singleUnitPrice"));
+      const lowestPrice = listingPrices.sort().at(0);
 
-      delete project.listings;
-
-      return { ...project, price };
+      return composeCarbonmarkProject(project, cmsProject, lowestPrice);
     });
-    const pooledProjects = pooledProjectsData.carbonOffsets.map(function (
-      project
-    ) {
-      /** Ignore projects for which the MARKETPLACE contained data */
-      if (project.display == false) {
-        return null;
-      }
-      const uniqueValues = [];
-
-      if (parseFloat(project.balanceUBO) >= 1) {
-        const isDefault =
-          POOL_INFO.ubo.defaultProjectTokenAddress.toLowerCase() ===
-          project.tokenAddress.toLowerCase();
-        const priceKey = isDefault ? "defaultPrice" : "selectiveRedeemPrice";
-        uniqueValues.push(poolPrices.ubo[priceKey]);
-      }
-      if (parseFloat(project.balanceNBO) >= 1) {
-        const isDefault =
-          POOL_INFO.nbo.defaultProjectTokenAddress.toLowerCase() ===
-          project.tokenAddress.toLowerCase();
-        const priceKey = isDefault ? "defaultPrice" : "selectiveRedeemPrice";
-        uniqueValues.push(poolPrices.nbo[priceKey]);
-      }
-      if (parseFloat(project.balanceNCT) >= 1) {
-        const isDefault =
-          POOL_INFO.nct.defaultProjectTokenAddress.toLowerCase() ===
-          project.tokenAddress.toLowerCase();
-        const priceKey = isDefault ? "defaultPrice" : "selectiveRedeemPrice";
-        uniqueValues.push(poolPrices.nct[priceKey]);
-      }
-      if (parseFloat(project.balanceBCT) >= 1) {
-        const isDefault =
-          POOL_INFO.bct.defaultProjectTokenAddress.toLowerCase() ===
-          project.tokenAddress.toLowerCase();
-        const priceKey = isDefault ? "defaultPrice" : "selectiveRedeemPrice";
-        uniqueValues.push(poolPrices.bct[priceKey]);
-      }
-
-      const country = project.country.length ? { id: project.country } : null;
 
-      const cmsData = findProjectWithRegistryIdAndRegistry(
+    // ----- Pool Listings ----- //
+    const offsetProjects = offsetData.carbonOffsets.map((offset) => {
+      const cmsProject = findProjectWithRegistryIdAndRegistry(
         projectsCmsData,
-        project.projectID.split("-")[1],
-        project.projectID.split("-")[0]
+        offset.projectID.split("-")[1], //e.g 1120
+        offset.projectID.split("-")[0] //e.g VCS
```

Comment:
can we use variables for these? it makes the comments obsolete and we only run `split` once:

```javascript
  const [firstName, lastName] = "John Doe".split(" ");
```

Reply:
Agreed, done